clients/githubrepo/client.go

```diff
@@ -47,27 +50,36 @@ func (client *Client) InitRepo(owner, repoName string) error {
 	url := client.repo.GetArchiveURL()
 	url = strings.Replace(url, "{archive_format}", "tarball/", 1)
 	url = strings.Replace(url, "{/ref}", client.repo.GetDefaultBranch(), 1)
-
 	req, err := http.NewRequestWithContext(client.ctx, http.MethodGet, url, nil)
 	if err != nil {
 		return fmt.Errorf("error during http.NewRequestWithContext: %w", err)
 	}
 	resp, err := http.DefaultClient.Do(req)
 	if err != nil {
-		return fmt.Errorf("%w", err)
+		return fmt.Errorf("error during HTTP call: %w", err)
 	}
 	defer resp.Body.Close()
 
-	respData, err := ioutil.ReadAll(resp.Body)
+	repoFile, err := os.OpenFile(repoFilename, os.O_CREATE|os.O_WRONLY, 0o644)
```

Comment:
how about storing this on disk unpacked as well? the disk space savings probably aren't worth the repeated unpack/decompressions that the checks are doing.

Reply:
Yes, that is on my mind too. Just don't want to attempt it now since changing this would require me to update `CheckIfFileExists` and `CheckFileContent`.

Eventually, I plan to expose `ListFiles()` and `GetFileContent()` APIs instead of a `GetRepoArchive()` API. When I make this change I will update the underlying data to be stored uncompressed.